contracts/src/v0.8/l2ep/dev/scroll/ScrollCrossDomainForwarder.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {TypeAndVersionInterface} from "../../../interfaces/TypeAndVersionInterface.sol";
+// solhint-disable-next-line no-unused-import
+import {ForwarderInterface} from "../interfaces/ForwarderInterface.sol";
+
+/* ./dev dependencies - to be moved from ./dev after audit */
+import {CrossDomainForwarder} from "../CrossDomainForwarder.sol";
+import {CrossDomainOwnable} from "../CrossDomainOwnable.sol";
+
+import {IScrollMessenger} from "@scroll-tech/contracts/libraries/IScrollMessenger.sol";
+import {Address} from "../../../vendor/openzeppelin-solidity/v4.7.3/contracts/utils/Address.sol";
+
+/**
+ * @title ScrollCrossDomainForwarder - L1 xDomain account representation
+ * @notice L2 Contract which receives messages from a specific L1 address and transparently forwards them to the destination.
+ * @dev Any other L2 contract which uses this contract's address as a privileged position,
+ *   can be considered to be owned by the `l1Owner`
+ */
+contract ScrollCrossDomainForwarder is TypeAndVersionInterface, CrossDomainForwarder {
+  // solhint-disable-next-line chainlink-solidity/prefix-immutable-variables-with-i
+  IScrollMessenger private immutable SCROLL_CROSS_DOMAIN_MESSENGER;
+
+  /**
+   * @notice creates a new Scroll xDomain Forwarder contract
+   * @param crossDomainMessengerAddr the xDomain bridge messenger (Scroll bridge L2) contract address
+   * @param l1OwnerAddr the L1 owner address that will be allowed to call the forward fn
+   */
+  constructor(IScrollMessenger crossDomainMessengerAddr, address l1OwnerAddr) CrossDomainOwnable(l1OwnerAddr) {
+    // solhint-disable-next-line custom-errors
```

Comment:
Do we need the require statements for backwards compatibility?

Reply:
I'm not too sure about the backwards compatibility portion (if any other reviewer has more insights on this please feel free to follow up here)

However, I don't think it hurts to include the additional check here (Optimism also had one as well). It also helps avoid errors on [this](https://github.com/smartcontractkit/chainlink/pull/11405/files#diff-c0e339e1a71f47f4392452bede111af3f7f5f6c6aa13444ca03bde7fc50e5800R82) line as well.

Reply:
I meant require over custom errors. We generally don't write new code that uses require statements

Reply:
@mohamed-mehany any thoughts here?

Reply:
@RensR Would it be possible to produce the same interface (including the errors) using custom errors? 
Because Optimism and Arbitrum L2EP validators also have the same error handling logic and it would be better if we can maintain the same interface across all chains contracts. 
If it's possible then we can do that in this PR, if not we can create a ticket to refactor all L2EP contracts (for all chains) to use custom errors.

Reply:
Yes but that completely defeats the purpose of custom errors. If you need it for backwards compatibility reasons, that valid. We can create a ticket to "modernize" all L2EP contracts, that would be great

Reply:
> We can create a ticket to "modernise" all L2EP contracts, that would be great

Sounds good! I will leave the require statements alone in this PR, and we can update them for the L2EP contracts at another time. @mohamed-mehany any preferences on where we should track the ticket? I believe the Scroll work is being tracked [here](https://smartcontract-it.atlassian.net/browse/DEPLOY-178), but not sure if you have another place in mind.